code/tailer/datastripe.q

```diff
@@ -118,33 +112,40 @@ upserttopartition:{[dir;tablename;keycol;enumdata;nextp]
 savetablesoverperiod:{[dir;tablename;nextp;lasttime]
     /- function to get keycol for table from access table
     keycol:`sym^.wdb.tablekeycols tablename;
-
     /- get distinct values to partition table on
     partitionlist: ?[tablename;();();(distinct;keycol)];
-
     /- enumerate and then split by keycol
     symdir:` sv dir,.proc.procname;
     enumkeycol: .Q.en[symdir;?[tablename;enlist (<;`time;nextp);0b;()]];
     splitkeycol: {[enumkeycol;keycol;s] ?[enumkeycol;enlist (=;keycol;enlist s);0b;()]}[enumkeycol;keycol;] each partitionlist;
-
     /-upsert table to partition
-    upserttopartition[dir;tablename;keycol;;nextp] each splitkeycol where 0<count each splitkeycol; 
-
+    upserttopartition[dir;tablename;keycol;;nextp] each splitkeycol where 0<count each splitkeycol;
     /- delete data from last period
-    .[.ds.deletetablebefore;(tablename;`time;lasttime)];
-    
+    ![;();0b;`symbol$()]each tablename; 
     /- run a garbage collection (if enabled)
     .gc.run[];
     };
 
-savealltablesoverperiod:{[dir;nextp;lasttime]
+savealltablesoverperiod:{[]
     /- function takes the tailer hdb directory handle and a timestamp
     /- saves each table up to given period to their respective partitions
-    savetablesoverperiod[dir;;nextp;lasttime]each .wdb.tablelist[];
+    /- totals calculates the row count of tables in .wdb.tablelist[] 
+    totals:{count get x}each .wdb.tablelist[];
+    /- If a totals value is greater than rowthresh which is the rowcount for saving down tables
+    if[max totals>.wdb.rowthresh;
```

Comment:
IMO this should be either

```suggestion
    if[max[totals]>.wdb.rowthresh;
```

or

```suggestion
    if[any totals>.wdb.rowthresh;
```

Technically these will all serve the same purpose, but I think it is semantically clearer to use `any` here

Reply:
This conditional has been removed, and the function has been restructured to only use one conditional